lib/TokenManager.ts

```diff
@@ -289,7 +289,7 @@ export class TokenManager {
   
     const onTokenExpiredHandler = (key) => {
       if (options.autoRenew) {
-        this.renew(key);
+        this.renew(key).catch(() => {}); // Renew errors will emit an "error" event 
```

Comment:
This sounds like this should get a CHANGELOG entry and a PENDING version bump at least as part of a patch level change (or minor if I'm following the "we don't do patch version bumps outside of hotfixes" conversation)

Reply:
This promise rejection was never catchable (notice the promise is not being returned to anyone). The "error" event is the only way to receive information about token renew failures which are done by this background process. This code catches and suppresses the "unhandled promise rejection"